scipy/stats/kde.py

```diff
@@ -197,7 +197,8 @@ def __init__(self, dataset, bw_method=None, weights=None):
         self.d, self.n = self.dataset.shape
 
         if weights is not None:
-            self._weights = atleast_1d(weights)
+            self._weights = weights.astype(float)
+            self._weights = atleast_1d(self._weights)
```

Comment:
This will break if weights is not a numpy array.

How about ``self._weights = atleast_1d(weights).astype(float)``?

Reply:
Yup, thanks. Fixed the order of operations, tweaked the test to cast `weights` being a list, and force-pushed.